### DISCUSSION TITLE
Q. 인수테스트의 시나리오를 어디까지 해야할까?

### DISCUSSION THREAD

**Discussion by jurlring** [woowacourse-teams/2022-gong-seek#86] [Polls]
ArticleAcceptance Test의 경우

1. 큰 사용자의 흐름
    1. 로그인을 하고 게시물을 등록한다.
    2. 로그인을 하지 않고 게시물을 등록할 수 없다.
    
    도메인에서 테스트 하므로 큰 흐름만 가져가자
    
2. 상세한 사용자의 흐름
    1. 제목이 없이 게시물을 등록할 수 없다.
    2. 내용이 1000자 이상이면 게시물을 등록할 수 없다.
    
    사용자의 관점에서 모든걸 테스트 하자

일단 1번처럼 구현하되, 시간이 남으면 2번처럼 해도 될 것 같다.

**Comment by brorae** [comment]
제목이나 내용 같은 부분은 도메인에서 테스트할 수 있을 것 같아요.
세세한 시나리오들은 기능 구현이 완료되고 나서 시간이 남으면 하는게 좋을 것 같습니다! 😄

**Comment by Gyuchool** [comment]
세세한 사항은 도메인테스트 혹은 다른 단위 테스트에서 검증한 것을 믿고 가고, 인수테스트는 전체적인 흐름을 테스트하는 방향으로 가는게 좋을 것 같다고 생각해요
